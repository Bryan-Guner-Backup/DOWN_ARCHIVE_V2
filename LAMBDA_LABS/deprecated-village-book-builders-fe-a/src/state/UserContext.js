import React, { useState, useContext, useEffect } from 'react';

const UserContext = React.createContext();

export function useUser() {
  return useContext(UserContext);
}

export function UserProvider({ children }) {
  const [userInfo, setUserInfo] = useState({
    username: localStorage.getItem('username'),
    role: localStorage.getItem('role'),
    id: localStorage.getItem('id'),
  });

  useEffect(() => {
    if (
      !localStorage.getItem('username') === null ||
      !localStorage.getItem('role') === null
    ) {
      return;
    } else {
      localStorage.setItem('username', userInfo.username);
      localStorage.setItem('role', userInfo.role);
      localStorage.setItem('id', userInfo.id);
    }
  }, [userInfo]);
  return (
    <UserContext.Provider value={{ userInfo, setUserInfo }}>
      {children}
    </UserContext.Provider>
  );
}
