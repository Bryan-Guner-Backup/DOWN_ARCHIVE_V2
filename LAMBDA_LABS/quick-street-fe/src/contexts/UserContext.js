import React, { createContext, useState } from 'react';

export const UserContext = createContext();

export const UserProvider = ({ children }) => {
  
  const [ user, setUser ] = useState({
    userId: '',
    isVendor: '',
    token: ''
  
  });
  
  const addUser = (newUser) => {
    // console.log('new user ', newUser);
    setUser({
      ...user, 
      userId: newUser.id, 
      token: newUser.token, 
      isVendor: newUser.isVendor
    });
  }

  return (
    <UserContext.Provider value={{ user, addUser }}>
      {children}
    </UserContext.Provider>
  )
}
