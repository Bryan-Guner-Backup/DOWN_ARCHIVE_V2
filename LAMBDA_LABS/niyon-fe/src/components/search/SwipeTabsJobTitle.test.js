import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import  SwipeTabsJobTitle  from './SwipeTabsJobTitle'
import { MemoryRouter } from 'react-router-dom'
import { UserContext } from '../../UserContext'
import findByTestAttr from '../../tests/utils'

describe('<SwipeTabsJobTitle />', () => {
   let wrapper
   const setUser = jest.fn()
   const user ={}
   const useStateSpy = jest.spyOn(React, 'useState')
   useStateSpy.mockImplementation((init) => [init, setUser])
   beforeEach(() => {
      wrapper = shallow(
        <UserContext.Provider value={{ user, setUser }}>
          <SwipeTabsJobTitle />
        </UserContext.Provider>
      )
   })

   it('should pass snapshot testing', () => {
        expect(toJSON(wrapper)).toMatchSnapshot();
   })
   it('should render <SwipeTabsJobTitle /> component correctly', () => {
        // console.log(wrapper.debug())
   })
})
