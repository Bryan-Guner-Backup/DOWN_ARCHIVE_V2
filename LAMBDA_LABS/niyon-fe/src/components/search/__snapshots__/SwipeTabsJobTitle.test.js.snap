// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SwipeTabsJobTitle /> should pass snapshot testing 1`] = `<SwipeTabsJobTitle />`;
